src/CMakeLists.txt

```diff
@@ -90,13 +96,12 @@ include(CTest)
 
 #create an object library for files that should be compiled with all test problems
 set (QuokkaObjSources "${CMAKE_CURRENT_SOURCE_DIR}/main.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/CloudyCooling.cpp" "${CMAKE_CURRENT_SOURCE_DIR}/GrackleDataReader.cpp")
-add_library(QuokkaObj OBJECT ${QuokkaObjSources})
-
-if(AMReX_GPU_BACKEND MATCHES "CUDA")
-    setup_target_for_cuda_compilation(QuokkaObj)
-endif()
-
-link_libraries(QuokkaObj)
+#we don't use it anymore because it gives issues on Cray systems
+#add_library(QuokkaObj OBJECT ${QuokkaObjSources} ${gamma_law_sources})
+#if(AMReX_GPU_BACKEND MATCHES "CUDA")
+#    setup_target_for_cuda_compilation(QuokkaObj)
+#endif()
+#link_libraries(QuokkaObj)
```

Comment:
Can you delete the commented-out code? It will still be in the git history, but it makes it cluttered to read.

Reply:
Yes, I thought about that. But I would like to keep it in the file for now until the Setonix build is successful. I know that it will be stored in the git history, but with so many PRs and commits in each PRs I lose track of which exact commit/PR was some info present in. For example, I worked on the `setup_target_for_microphysics_compilation` function a while ago, and it took some man hours to dig the right PR/commit out. And these extra linking/de-linking commands took me hours to work through.

I want to revisit this object library issue once we build it fine on Cray and Setonix. There is also some other old commented out code in this file, so we can make a quick, separate PR later on to de clutter the file